From 9a09a4848bd3e2fabcb875b381ce09dd4594f532 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Mon, 6 Feb 2017 20:24:52 +0900
Subject: [PATCH 1808/2066] clk: renesas: rcar-gen3: Fix SD divider setting

This patch fixed the SD divider settiing for corresponding to the change
in the HS200/HS400 mode.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/clk/renesas/rcar-gen3-cpg.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/clk/renesas/rcar-gen3-cpg.c b/drivers/clk/renesas/rcar-gen3-cpg.c
index 31246ab..f49046e 100644
--- a/drivers/clk/renesas/rcar-gen3-cpg.c
+++ b/drivers/clk/renesas/rcar-gen3-cpg.c
@@ -1,6 +1,7 @@
 /*
  * R-Car Gen3 Clock Pulse Generator
  *
+ * Copyright (C) 2017 Renesas Electronics Corp.
  * Copyright (C) 2015-2016 Glider bvba
  *
  * Based on clk-rcar-gen3.c
@@ -494,12 +495,12 @@ struct sd_clock {
  *                     sd_srcfc   sd_fc   div
  * stp_hck   stp_ck    (div)      (div)     = sd_srcfc x sd_fc
  *-------------------------------------------------------------------
- *  0         0         1 (2)      0 (2)      4 : SDR104 / HS400 / HS200
+ *  0         0         1 (2)      0 (-)      2 : HS400
+ *  0         0         0 (1)      1 (4)      4 : SDR104 / HS200
  *  0         0         1 (2)      1 (4)      8 : SDR50
  *  1         0         2 (4)      1 (4)     16 : HS / SDR25
  *  1         0         3 (8)      1 (4)     32 : NS / SDR12
  *  0         0         0 (1)      0 (2)      2 : (no case)
- *  0         0         0 (1)      1 (4)      4 : (no case)
  *  1         0         2 (4)      0 (2)      8 : (no case)
  *  1         0         3 (8)      0 (2)     16 : (no case)
  *  1         0         4 (16)     0 (2)     32 : (no case)
@@ -507,12 +508,12 @@ struct sd_clock {
  */
 static const struct sd_div_table cpg_sd_div_table[] = {
 /*	CPG_SD_DIV_TABLE_DATA(stp_hck,  stp_ck,   sd_srcfc,   sd_fc,  sd_div) */
-	CPG_SD_DIV_TABLE_DATA(0,	0,	  1,	      0,        4),
+	CPG_SD_DIV_TABLE_DATA(0,	0,	  1,	      0,        2),
+	CPG_SD_DIV_TABLE_DATA(0,	0,	  0,	      1,        4),
 	CPG_SD_DIV_TABLE_DATA(0,	0,	  1,	      1,        8),
 	CPG_SD_DIV_TABLE_DATA(1,	0,	  2,	      1,       16),
 	CPG_SD_DIV_TABLE_DATA(1,	0,	  3,	      1,       32),
 	CPG_SD_DIV_TABLE_DATA(0,	0,	  0,	      0,        2),
-	CPG_SD_DIV_TABLE_DATA(0,	0,	  0,	      1,        4),
 	CPG_SD_DIV_TABLE_DATA(1,	0,	  2,	      0,        8),
 	CPG_SD_DIV_TABLE_DATA(1,	0,	  3,	      0,       16),
 	CPG_SD_DIV_TABLE_DATA(1,	0,	  4,	      0,       32),
-- 
1.7.5.4

